Sources/TuistKit/Models/Up/UpHomebrewTap.swift

```diff
@@ -0,0 +1,98 @@
+import Basic
+import Foundation
+import TuistCore
+
+/// Command that installs Homebrew and packages.
+class UpHomebrewTap: Up, GraphInitiatable {
+    /// Up name.
+    static let name: String = "Homebrew tap"
+
+    /// Tap repositories
+    let repositories: [String]
+
+    /// Up homebrew for installing Carthge.
+    let upHomebrew: Upping
+
+    /// Initializes the Homebrew tap up.
+    ///
+    /// - Parameter
+    ///     - repositories: List of repositories to be tapped.
+    ///   - upHomebrew: Up homebrew for installing Homebrew.
+    init(repositories: [String],
+         upHomebrew: Upping = UpHomebrew(packages: [])) {
+        self.repositories = repositories
+        self.upHomebrew = upHomebrew
+        super.init(name: UpHomebrewTap.name)
+    }
+
+    /// Default constructor of entities that are part of the manifest.
+    ///
+    /// - Parameters:
+    ///   - dictionary: Dictionary with the object representation.
+    ///   - projectPath: Absolute path to the folder that contains the manifest.
+    ///     This is useful to obtain absolute paths from the relative paths provided in the manifest by the user.
+    ///   - fileHandler: File handler for any file operations like checking whether a file exists or not.
+    /// - Throws: A decoding error if an expected property is missing or has an invalid value.
+    required convenience init(dictionary: JSON, projectPath _: AbsolutePath, fileHandler _: FileHandling) throws {
+        self.init(repositories: try dictionary.get("repositories"))
+    }
+
+    /// Returns true when the command doesn't need to be run.
+    ///
+    /// - Parameters
+    ///   - system: System instance to run commands on the shell.
+    ///   - projectPath: Path to the directory that contains the project manifest.
+    /// - Returns: True if the command doesn't need to be run.
+    /// - Throws: An error if the check fails.
+    override func isMet(system: Systeming, projectPath: AbsolutePath) throws -> Bool {
+        if try !upHomebrew.isMet(system: system, projectPath: projectPath) { return false }
+        let taps = try self.taps(system: system)
+        for repository in repositories {
+            if !isTapConfigured(repository, taps: taps) {
+                return false
```

Comment:
Using a for-loop like this makes it harder to skim read, It might be better to use `repositories.first(where:)` to better show our intent here.